src/ascii/interger.cairo

```diff
@@ -0,0 +1,222 @@
+use core::option::OptionTrait;
+use alexandria::data_structures::array_ext::ArrayTraitExt;
+use array::{ArrayTrait, Array};
+use traits::{Into, TryInto, DivRem};
+use zeroable::Zeroable;
+
+trait IntergerToAsciiTrait<T, U> {
+    fn to_ascii(self: T) -> U;
+}
+
+// converts intergers into an array of its individual ascii values
+trait IntergerToAsciiArrayTrait<T> {
+    fn to_ascii_array(self: T) -> Array<felt252>;
+    fn to_inverse_ascii_array(self: T) -> Array<felt252>;
+}
+
+// converts intergers into an array of its individual ascii values
+// e.g. 123 -> [49, 50, 51]
+impl IntergerToAsciiArrayTraitImpl<
+    T,
+    impl TNumericLiteral: NumericLiteral<T>,
+    impl TPartialOrd: PartialOrd<T>,
+    impl TDivRem: DivRem<T>,
+    impl TInto: Into<T, felt252>,
+    impl TryInto: TryInto<felt252, T>,
+    impl TTryIntoZero: TryInto<T, NonZero<T>>,
+    impl TZeroable: Zeroable<T>,
+    impl TDrop: Drop<T>,
+    impl TCopy: Copy<T>,
+> of IntergerToAsciiArrayTrait<T> {
+    fn to_ascii_array(self: T) -> Array<felt252> {
+        let mut new_arr = self.to_inverse_ascii_array();
+        new_arr.reverse();
+        new_arr
+    }
+
+    fn to_inverse_ascii_array(self: T) -> Array<felt252> {
+        let mut new_arr = ArrayTrait::new();
+        if self <= 9.try_into().unwrap() {
+            new_arr.append(self.into() + 48);
+            return new_arr;
+        }
+
+        let mut num = self;
+        loop {
+            if num.is_zero() {
+                break ();
+            }
+            let (quotient, remainder) = DivRem::div_rem(
+                num, TryInto::<felt252, T>::try_into(10).unwrap().try_into().expect('Division by 0')
```

Comment:
missing a try_into here :)

Reply:
i cant seem to use try_into() like that without specifying the type as it fails to infer the trait

Reply:
you prob can force the type by doing `10.try_into::<felt, T>().unwrap()`

Reply:
hmm dosent seem to work the error is `Wrong number of generic arguments. Expected 0, found: 2`